app/controllers/search_controller.rb

```diff
@@ -1,8 +1,18 @@
 class SearchController < ApplicationController
+  include SearchHelper
   def search
     session[:address] = params[:search] if params.has_key?(:search)
-    @search = Listing.where(is_active: true)
-    @search = @search.near("#{session[:address]}", 5, order: 'distance')
+    if check_if_filled?(session[:address])
+      @search = Listing.where(is_active: true)
+    else
+      @search = Listing.where(is_active: true).near("#{session[:address]}", 5, order: 'distance')
+    end
+
+    if check_if_filled?(params[:check_in_date]) && check_if_filled?(params[:check_out_date])
+      @bookings = Booking.where("'#{params[:check_in_date]}' < check_in_date  AND check_out_date < '#{params[:check_out_date]}'")
+                        .pluck(:listing_id)
+      @search = @search.where.not(id: @bookings)
+    end
     @search = @search.ransack(params[:q])
     @listing_list = @search.result(distinct: true).paginate(page: params[:page], per_page: 2)
```

Comment:
Before going here, check if `@search` has contents.

There was an issue with the search method, where Listings that are very far from the address being searched still appear under the results.

I checked your code (with fixes on the `params[:search]` tag and `session[:address]`) and found the following:
<img width="1792" alt="Screen Shot 2021-04-28 at 6 52 57 PM" src="https://user-images.githubusercontent.com/20994464/116393171-e0ab3280-a853-11eb-88fe-18974c205711.png">
`@search` is empty. This is correct, because the place that I'm searching for is in "Japan" and I don't have any records with that address.

With your code, it still proceeds to the next step even though `@search` is empty.